File: ProbSolv/sherMin.py

def sher_min(arr,n,p,q):
    arr1 = []
    cool = []
    for i in range(p,q+1):
        arr2 = []
        for j in range(n):
            arr2.append(abs(arr[j]-i))
        cool.append(min(arr2))
        arr1.append(i)
    
    dude = max(cool)
    ind = cool.index(dude)
    #print cool
    #print arr1
    print arr1[ind]
    


n = int(raw_input())
a = raw_input().split()
a = [int(x) for x in a]
p_q = raw_input().split()
p_q = [int(x) for x in p_q]
p = p_q[0]
q = p_q[1]

sher_min(a,n,p,q)

File: ProbSolv/sherMin.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<cmath>
using namespace std;
long long sher_min(vector <long long> arr,int n,int p,int q)
{
	vector<long long> vec1;
	vector<long long> cool;

	for (int i = p; i<=q; ++i)
	{
		vector<long long> vec2;
		for (int j = 0; j<n; ++j)
		{
			vec2.push_back(abs(arr[j]-i));
		}
		cool.push_back(*min_element(vec2.begin(), vec2.end()));
		vec1.push_back(i);
	}
	long long dude = *max_element(cool.begin(),cool.end());
	long long index = distance(cool.begin(), find(cool.begin(),cool.end(),dude));
	return vec1[index];
}


int main()
{

	int n;
	cin >> n;
	//cout << endl;
	vector<long long> v;
	for (int i = 0; i<n; ++i)
	{
		long long dude;
		cin >> dude;
		v.push_back(dude);
	}
	vector<int>::iterator it;
	/*
	for (it = v.begin(); it<v.end(); ++it)
	{
		cout << *it << " ";
	}
	
	cout << endl;
	*/
	long long p,q;
	cin >> p >> q;
	//cout << p << endl;
	//cout << q << endl;
	cout << sher_min(v,n,p,q) << endl;
}
